Add docs for changing cluster plan between Standard and Basic

katmayb opened the pull request:
Fixes DOC-10627

This PR adds a page to describe changing cluster plans between Standard and Basic clusters.

Also:

- Added a short section on the cluster management pages for Basic and Standard to direct customers to the new page.
- Updated some errors here and there.

## Preview

New page: https://deploy-preview-19462--cockroachdb-docs.netlify.app/docs/cockroachcloud/change-plan-between-basic-and-standard.html

jaiayu commented on src/current/cockroachcloud/basic-cluster-management.md:
```diff
@@ -85,6 +85,12 @@ A CockroachDB {{ site.data.products.basic }} cluster cannot be moved from one re
+
+To change your cluster's plan between {{ site.data.products.basic }} and {{ site.data.products.standard }}, refer to [Change a Cluster's Plan Between Standard and Basic]({% link cockroachcloud/change-plan-between-basic-and-standard.md %}).
+
+To change from {{ site.data.products.basic }} to {{ site.data.products.advanced }}, you must first change your cluster's plan from {{ site.data.products.basic }} to {{ site.data.products.standard }}, and then you can refer to [Change a Cluster's Plan from Standard to Advanced]({% link cockroachcloud/change-plan-from-standard-to-advanced.md %}).
```
I am not sure this is required - I think you would be able to go straight from Basic to Advanced with these instructions. Can you check with @marksoper or @devarshishah3?

jaiayu commented on src/current/cockroachcloud/change-plan-between-basic-and-standard.md:
```diff
@@ -0,0 +1,39 @@
+
+## Before you begin
+
+You'll need the [Cluster Administrator]({% link cockroachcloud/authorization.md %}#cluster-administrator) role on the running {{ site.data.products.standard }} or {{ site.data.products.basic }} cluster in order to change the cluster's [plan]({% link cockroachcloud/index.md %}#plans).
```
@katmayb Cluster Operator roles can also perform the plan change.

katmayb replied:
done!

jaiayu commented on src/current/cockroachcloud/change-plan-between-basic-and-standard.md:
```diff
@@ -0,0 +1,39 @@
+1. On the [**Cluster Overview** page]({% link cockroachcloud/cluster-overview-page.md %}), select the **Action** dropdown in the top-right corner, and then **Change plan type**.
+1. Choose a [plan]({% link cockroachcloud/index.md %}#plans) from {{ site.data.products.standard }} or {{ site.data.products.basic }} to change to. The current plan of the cluster will be highlighted. Select **Next: Capacity**.
+1. For capacity:
+    - If you are changing to a {{ site.data.products.standard }} plan, choose the amount of [provisioned capacity]({% link cockroachcloud/plan-your-cluster.md %}#provisioned-capacity) you want for your cluster. Select **Next: Finalize**.
```
@katmayb we give a recommendation in CC console for your provisioned vCPU depending on your peak RU/s in the last 30 days. For your reference, we calculate this vCPU based on the 30-day peak RU/s plus a 40% buffer, then convert using 500 RU/s = 1 vCPU. We use a 10-minute roll-up for the peak RU/s. -- This calculation may be good to include in docs somewhere to reference; Mike had suggested adding it to "Plan your cluster".

There are also a few caveats that we may want to mention with respect to the recommendation in the UI: 
- This should be interpreted as a suggestion. Users should keep an eye on their CPU utilization % and adjust their compute capacity as needed. 
- Standard currently supports a maximum of 60 vCPUs. If we estimate you need more than 60 vCPU, we tell them that we estimate they need more than our capacity, but "You can provision up to 60 vCPU. If your workload consistently spikes above that level you may experience performance issues."

katmayb replied:
I added a section to "plan your cluster" with the calculation. And I've added that the Cloud Console provides a suggestion but that users should monitor this.

Your last bullet point — I've added to the Plan your cluster page. I haven't not specifically mentioned the 60 vCPUs for ease of keeping the docs up to date. But, if that's a number that won't change frequently or you'd prefer it specified I can update these changes.

jaiayu commented on src/current/cockroachcloud/change-plan-between-basic-and-standard.md:
```diff
@@ -0,0 +1,39 @@
+1. For capacity:
+    - If you are changing to a {{ site.data.products.standard }} plan, choose the amount of [provisioned capacity]({% link cockroachcloud/plan-your-cluster.md %}#provisioned-capacity) you want for your cluster. Select **Next: Finalize**.
+    - If you are changing to a [{{ site.data.products.basic }} plan]({% link cockroachcloud/create-a-basic-cluster.md %}#step-4-configure-cluster-capacity), choose whether the cluster should have **Unlimited** capacity to scale, or **Set a monthly limit**. Select **Next: Finalize**.
+1. Review the changes and then **Update cluster**.
```
We include the following in the UI, if it makes sense to include here as well: 

- The upgrade process will take up to 5 mins
- Their cluster will remain functional under the current plan during the switching period
- They can switch back the plan if needed. But when you go from Standard to Basic, you need to wait 48 hours before being able to change the plan back to Standard.

katmayb replied:
The 48 hours wait is just for if the cluster was originally Standard? I have a note earlier on the page that mentions this generally about any switch back, so maybe I need to change that?

I have added these as bullet points to the final step.

jaiayu commented on src/current/cockroachcloud/change-plan-from-standard-to-advanced.md:
```diff
@@ -9,23 +9,23 @@ docs_area: migrate
 
-- A [CockroachDB {{ site.data.products.standard }} or {{ site.data.products.basic }} cluster]({% link cockroachcloud/quickstart.md %}) from which you want to migrate data.
+- A [CockroachDB {{ site.data.products.standard }} cluster]({% link cockroachcloud/quickstart.md %}) of which you want to change the plan.
 - A [paid CockroachDB {{ site.data.products.advanced }} cluster]({% link cockroachcloud/quickstart-trial-cluster.md %}). Your first paid CockroachDB {{ site.data.products.advanced }} cluster is free for a 30-day trial.
```
Line 21 - the part about your first cluster being free for 30 days is actually no longer true. We have replaced this version of the free trial now.

katmayb replied:
Removed

jaiayu reviewed (changes requested):
Thanks @katmayb. I left a few comments inline. There are a couple of additional considerations that should be called out for the UI workflow:
- If the cluster was created on legacy architecture, then it cannot be upgraded via UI. The cluster will not support multi-region features or private connections. To use these features, a customer would need to do backup and restore into a new cluster. If they still want to change the plan for their current cluster, they can do so via API (https://www.cockroachlabs.com/docs/cockroachcloud/cloud-api#change-a-clusters-plan).
- If a cluster was recently downgraded to Basic from Standard, it must wait 48 hours to be upgraded back to Standard.
- When you're downgrading from Standard to Basic, you need to disable Standard-only premium features first (private connections, manual version control setting, metrics/log export to DataDog/Prometheus). 

I also spoke to Devarshi about changes to the "Migrate from Standard/Basic to Advanced" page. He said we should keep using the terminology "Migrate" in that case since we don't support simple plan changes for that flow yet. The user would still need to perform a full migration.

katmayb commented:
> Thanks @katmayb. I left a few comments inline. There are a couple of additional considerations that should be called out for the UI workflow:
> 
> * If the cluster was created on legacy architecture, then it cannot be upgraded via UI. The cluster will not support multi-region features or private connections. To use these features, a customer would need to do backup and restore into a new cluster. If they still want to change the plan for their current cluster despite the feature limitations, they can do so via API (https://www.cockroachlabs.com/docs/cockroachcloud/cloud-api#change-a-clusters-plan).
> * If a cluster was recently downgraded to Basic from Standard, it must wait 48 hours to be upgraded back to Standard.
> * When you're downgrading from Standard to Basic, you need to disable Standard-only premium features first (private connections, manual version control setting, metrics/log export to DataDog/Prometheus).
> 
> I also spoke to Devarshi about changes to the "Migrate from Standard/Basic to Advanced" page. He said we should keep using the terminology "Migrate" in that case since we don't support simple plan changes for that flow yet. The user would still need to perform a full migration.

I have mentioned these in the "Considerations" and prerequisites. Regarding "legacy architecture", what do we mean here exactly — and is that how we want to phrase it in docs?

jaiayu commented on src/current/cockroachcloud/basic-cluster-management.md:
```diff
@@ -85,6 +85,12 @@ A CockroachDB {{ site.data.products.basic }} cluster cannot be moved from one re
+
+To change your cluster's plan between {{ site.data.products.basic }} and {{ site.data.products.standard }}, refer to [Change a Cluster's Plan Between Standard and Basic]({% link cockroachcloud/change-plan-between-basic-and-standard.md %}).
+
+To change from {{ site.data.products.basic }} to {{ site.data.products.advanced }}, you must first change your cluster's plan from {{ site.data.products.basic }} to {{ site.data.products.standard }}, and then you can refer to [Change a Cluster's Plan from Standard to Advanced]({% link cockroachcloud/migrate-from-standard-to-advanced.md %}).
```
@katmayb can we remove "you must first change your cluster's plan from" -- I think they can use the Standard <> Advanced instructions to directly migrate from Basic to Advanced.

katmayb replied:
Yes, sorry! Updated.

jaiayu commented on src/current/cockroachcloud/change-plan-between-basic-and-standard.md:
```diff
@@ -0,0 +1,48 @@
+## Change a cluster's plan
+
+{{site.data.alerts.callout_info}}
+After changing your cluster's plan, you will not be able to change the plan again for 48 hours.
```
This is only true for Standard to Basic, like you have it called out under considerations

katmayb replied:
I have updated this to include the Standard to Basic callout.

jaiayu commented on src/current/cockroachcloud/plan-your-cluster.md:
```diff
@@ -43,6 +43,10 @@ You can decrease the provisioned capacity only three times within a 7-day period
 
+### Changing cluster plan provisioned capacity recommendations
+
+If you change your cluster's plan from {{ site.data.products.basic }} to {{ site.data.products.standard }}, the Cloud Console will provide a suggested provisioned capacity. This is calculated depending on your [peak request units (RUs)]({% link cockroachcloud/plan-your-cluster-basic.md %}) in the last 30 days on the {{ site.data.products.basic }} cluster plan. The vCPU suggestion is based on the  ...
```
This is great. Two comments: 
- The unit we look at is peak RU per second. So anywhere we are referencing RUs -- that should be RU/sec.
- The last thing we might want to include is the roll-up period we use for the peak RU/sec. We use a 10-minute roll up period.

jaiayu commented on src/current/cockroachcloud/basic-cluster-management.md:
```diff
@@ -85,6 +85,12 @@ A CockroachDB {{ site.data.products.basic }} cluster cannot be moved from one re
+
+To change your cluster's plan between {{ site.data.products.basic }} and {{ site.data.products.standard }}, refer to [Change a Cluster's Plan Between Standard and Basic]({% link cockroachcloud/change-plan-between-basic-and-standard.md %}).
+
+To change from {{ site.data.products.basic }} to {{ site.data.products.advanced }}, refer to [Change a Cluster's Plan from Standard or Basic to Advanced]({% link cockroachcloud/migrate-from-standard-to-advanced.md %}).
```
I think the title needs to be updated here - still says "Change a cluster's plan" instead of "Migrate..."

katmayb replied:
Done!

jaiayu commented on src/current/cockroachcloud/cluster-management.md:
```diff
@@ -106,11 +106,17 @@ When the upgrade occurs, your cluster status will be listed as `Available (Maint
+
+To change your cluster's plan between {{ site.data.products.basic }} and {{ site.data.products.standard }}, refer to [Change a Cluster's Plan Between Standard and Basic]({% link cockroachcloud/change-plan-between-basic-and-standard.md %}).
+
+To change from {{ site.data.products.standard }} to {{ site.data.products.advanced }}, refer to [Change a Cluster's Plan from Standard to Advanced]({% link cockroachcloud/migrate-from-standard-to-advanced.md %}).
```
Same comment as above. The title of the page might need to be updated.

katmayb replied:
Done!

jaiayu reviewed (commented):
Can we also update cloud release notes @katmayb! Thank you!

katmayb commented:
> Can we also update cloud release notes @katmayb! Thank you!

@jaiayu I've added a very simple cloud release note. I added yesterday's date, not sure when the release date is. Please let me know! Changed the instances of RUs to RU/sec (we may need to do a follow-up to check this across the docs, I noticed inconsistency here).

jaiayu commented on src/current/releases/cloud.md:
```diff
@@ -14,6 +14,10 @@ Get future release notes emailed to you:
 
 {% include marketo.html formId=1083 %}
 
+## March 25, 2025
```
This needs to be updated to April 3.

florence-crl commented on src/current/cockroachcloud/change-plan-between-basic-and-standard.md:
```diff
@@ -0,0 +1,48 @@
+
+1. On the [**Clusters** page]({% link cockroachcloud/cluster-management.md %}#view-clusters-page), select the cluster that requires a plan change.
+1. On the [**Cluster Overview** page]({% link cockroachcloud/cluster-overview-page.md %}), select the **Action** dropdown in the top-right corner, and then **Change plan type**.
+1. Choose a [plan]({% link cockroachcloud/index.md %}#plans) from {{ site.data.products.standard }} or {{ site.data.products.basic }} to change to. The current plan of the cluster will be highlighted. Select **Next: Capacity**.
```
<img width="807" alt="Screenshot 2025-04-03 at 2 46 10 PM" src="https://github.com/user-attachments/assets/2358c54b-e2f8-4aad-a548-f0348e3f6bf6" />

```suggestion
1. Choose a [plan]({% link cockroachcloud/index.md %}#plans) from {{ site.data.products.standard }} or {{ site.data.products.basic }} to change to. The current plan of the cluster will be labeled, while the other plan type will be highlighted. Select **Next: Capacity**.
```

florence-crl commented on src/current/cockroachcloud/change-plan-between-basic-and-standard.md:
```diff
@@ -0,0 +1,48 @@
+1. For capacity:
+    - If you are changing to a {{ site.data.products.standard }} plan, choose the amount of [provisioned capacity]({% link cockroachcloud/plan-your-cluster.md %}#provisioned-capacity) you want for your cluster. The Cloud Console provides a suggestion for provisioned vCPU, we recommend monitoring CPU utilization and adjusting the cluster's compute capacity as necessary. For additional information, ...
+    - If you are changing to a [{{ site.data.products.basic }} plan]({% link cockroachcloud/create-a-basic-cluster.md %}#step-4-configure-cluster-capacity), choose whether the cluster should have **Unlimited** capacity to scale, or **Set a monthly limit**. Select **Next: Finalize**.
+1. Review the changes and then **Update cluster**. Consider the following:
```
```suggestion
1. Review the changes and then **Update cluster**.
```
Please either remove "consider the following" or add considerations.

florence-crl reviewed (approved):
lgtm pending suggestions.

florence-crl commented on src/current/cockroachcloud/plan-your-cluster.md:
```diff
@@ -43,6 +43,10 @@ You can decrease the provisioned capacity only three times within a 7-day period
 
 The CockroachDB {{ site.data.products.cloud }} Console provides insight into your cluster's compute usage relative to its provisioned capacity on a monthly basis. We recommend that you maintain at minimum a 40% capacity buffer over time, and that you increase this buffer if the workload is unpredictable or highly sensitive to query latency. You now have the advantage of historical data about the  ...
 
+### Changing cluster plan provisioned capacity recommendations
```
one last suggestion, would you be able to change this heading to: "Recommended provisioned capacity when changing cluster plans"?

katmayb replied:
```suggestion
### Recommended provisioned capacity when changing cluster plan
```

katmayb commented on src/current/cockroachcloud/change-plan-between-basic-and-standard.md:
```diff
@@ -0,0 +1,48 @@
+{{site.data.alerts.end}}
+
+1. On the [**Clusters** page]({% link cockroachcloud/cluster-management.md %}#view-clusters-page), select the cluster that requires a plan change.
+1. On the [**Cluster Overview** page]({% link cockroachcloud/cluster-overview-page.md %}), select the **Action** dropdown in the top-right corner, and then **Change plan type**.
```
```suggestion
1. On the [**Cluster Overview** page]({% link cockroachcloud/cluster-overview-page.md %}), select the **Action** dropdown in the top-right corner, and then **Change plan type**. If this option is unavailable, refer to [Considerations](#considerations).
```

katmayb commented on src/current/releases/cloud.md:
```diff
@@ -14,6 +14,10 @@ Get future release notes emailed to you:
 
 {% include marketo.html formId=1083 %}
 
+## April 4, 2025
```
```suggestion
## April 3, 2025
```

katmayb commented on src/current/cockroachcloud/change-plan-between-basic-and-standard.md:
```diff
@@ -0,0 +1,48 @@
+1. On the [**Cluster Overview** page]({% link cockroachcloud/cluster-overview-page.md %}), select the **Action** dropdown in the top-right corner, and then **Change plan type**. If this option is unavailable, refer to [Considerations](#considerations).
+1. Choose a [plan]({% link cockroachcloud/index.md %}#plans) from {{ site.data.products.standard }} or {{ site.data.products.basic }} to change to. The current plan of the cluster will be labeled, while the other plan type will be highlighted. Select **Next: Capacity**.
+1. For capacity:
+    - If you are changing to a {{ site.data.products.standard }} plan, choose the amount of [provisioned capacity]({% link cockroachcloud/plan-your-cluster.md %}#provisioned-capacity) you want for your cluster. The Cloud Console provides a suggestion for provisioned vCPU, we recommend monitoring CPU utilization and adjusting the cluster's compute capacity as necessary. For additional information, ...
```
```suggestion
    - If you are changing to a {{ site.data.products.standard }} plan, choose the amount of [provisioned capacity]({% link cockroachcloud/plan-your-cluster.md %}#provisioned-capacity) you want for your cluster. The Cloud Console provides a suggestion for provisioned vCPU, we recommend monitoring CPU utilization and adjusting the cluster's compute capacity as necessary. For additional information, refer to [Recommended provisioned capacity when changing cluster plan]({% link cockroachcloud/plan-your-cluster.md %}#recommended-provisioned-capacity-when-changing-cluster-plan). Select **Next: Finalize**.
```
